src/Domain/CoordinateMaps/Frustum.hpp

```diff
@@ -51,8 +52,11 @@ class Frustum {
   std::array<tt::remove_cvref_wrap_t<T>, 3> operator()(
       const std::array<T, 3>& source_coords) const noexcept;
 
+  /// Returns invalid if \f$z\f$ is at or beyond the \f$z\f$-coordinate of
+  /// the apex of the pyramid formed by extending the `Frustum` (for a
```

Comment:
Extended Frustums only end a pyramid if one of the bases of the frustum is smaller in both directions than the other base. In cases where the bases are the same length along one direction, the extended frustum makes a triangular prism. A third case where one base is longer in one direction and shorter in the second direction leads to a tetrahedron instead. 

But I think this can be reworded such that it works for all cases, since these are all triangles in the x-z and y-z planes, and the apex of the triangle in the plane is what will determine where the mapping becomes singular.